gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java

```diff
@@ -403,41 +453,58 @@ public EntityPlayer teleport(Location to, EntityTeleportEvent.Cause cause) {
         // Reset chunks
         this.connection.resetQueuedChunks();
 
+        Chunk fromChunk = this.chunk();
+
         // Check if we need to change worlds
         if (!to.world().equals(from.world())) {
+            if (this.world != null) {
+                this.world.server().pluginManager().callEvent(new PlayerWorldLeaveEvent(this));
+            }
             // Despawn entities first
             this.entityVisibilityManager.clear();
 
             // Change worlds
             world().removePlayer(this);
             this.world((WorldAdapter) to.world());
```

Comment:
This needs to be done in to.world().syncScheduler().execute()

Reply:
done